controlplane/pkg/prefix_pool/prefixpool.go

```diff
@@ -47,14 +49,15 @@ func NewPrefixPool(prefixes ...string) (PrefixPool, error) {
 	}, nil
 }
 
-func (impl *prefixPool) Extract(connectionId string, family connectioncontext.IpFamily_Family, requests ...*connectioncontext.ExtraPrefixRequest) (srcIP *net.IPNet, dstIP *net.IPNet, requested []string, err error) {
+func (impl *prefixPool) Extract(connectionId string, family connectioncontext.IpFamily_Family, excludedPrefixes []string, requests ...*connectioncontext.ExtraPrefixRequest) (srcIP *net.IPNet, dstIP *net.IPNet, requested []string, err error) {
```

Comment:
Probably it's better to create new method
```
ExtractWithExcludedPrefix(connectionId string, family connectioncontext.IpFamily_Family, excludedPrefixes []string, requests ...*connectioncontext.ExtraPrefixRequest) (srcIP *net.IPNet, dstIP *net.IPNet, requested []string, err error)
```
And leave `Extract` as it is:
```
Extract(connectionId string, family connectioncontext.IpFamily_Family, requests ...*connectioncontext.ExtraPrefixRequest) (srcIP *net.IPNet, dstIP *net.IPNet, requested []string, err error) {
    return ExtractWithExcludedPrefix(connectionId, family, []string{}, requests...)
}
```

Reply:
@rdimitrov, please share your approach here. Why is it better?